src/pages/RoomInvitePage.tsx

```diff
@@ -184,60 +160,51 @@ function RoomInvitePage(props) {
+    const reportName = useMemo(() => ReportUtils.getReportName(report), [report]);
     const inviteUsers = useCallback(() => {
         if (!validate()) {
             return;
         }
-        const invitedEmailsToAccountIDs = {};
-        _.each(selectedOptions, (option) => {
-            const login = option.login || '';
-            const accountID = lodashGet(option, 'accountID', '');
+        const invitedEmailsToAccountIDs: PolicyUtils.MemberEmailsToAccountIDs = {};
+        selectedOptions.forEach((option) => {
+            const login = option.login ?? '';
+            const accountID = option.accountID;
             if (!login.toLowerCase().trim() || !accountID) {
                 return;
             }
             invitedEmailsToAccountIDs[login] = Number(accountID);
         });
-        Report.inviteToRoom(props.report.reportID, invitedEmailsToAccountIDs);
+        if (reportID) {
+            Report.inviteToRoom(reportID, invitedEmailsToAccountIDs);
+        }
         Navigation.navigate(backRoute);
-    }, [selectedOptions, backRoute, props.report.reportID, validate]);
+    }, [selectedOptions, backRoute, reportID, validate]);
 
     const headerMessage = useMemo(() => {
         const searchValue = searchTerm.trim().toLowerCase();
-        if (!userToInvite && CONST.EXPENSIFY_EMAILS.includes(searchValue)) {
+        const expensifyEmails = CONST.EXPENSIFY_EMAILS as string[];
+        if (!userToInvite && expensifyEmails.includes(searchValue)) {
             return translate('messages.errorMessageInvalidEmail');
         }
-        if (
-            !userToInvite &&
-            excludedUsers.includes(
-                parsePhoneNumber(LoginUtils.appendCountryCode(searchValue)).possible ? OptionsListUtils.addSMSDomainIfPhoneNumber(LoginUtils.appendCountryCode(searchValue)) : searchValue,
-            )
-        ) {
+        if (!userToInvite && excludedUsers.includes(searchValue)) {
             return translate('messages.userIsAlreadyMember', {login: searchValue, name: reportName});
         }
-        return OptionsListUtils.getHeaderMessage(personalDetails.length !== 0, Boolean(userToInvite), searchValue);
-    }, [excludedUsers, translate, searchTerm, userToInvite, personalDetails, reportName]);
+        return OptionsListUtils.getHeaderMessage(invitePersonalDetails.length !== 0, Boolean(userToInvite), searchValue);
+    }, [excludedUsers, translate, searchTerm, userToInvite, invitePersonalDetails, reportName]);
     return (
         <ScreenWrapper
             shouldEnableMaxHeight
             testID={RoomInvitePage.displayName}
         >
             <FullPageNotFoundView
-                shouldShow={_.isEmpty(props.report)}
-                subtitleKey={_.isEmpty(props.report) ? undefined : 'roomMembersPage.notAuthorized'}
+                shouldShow={!report}
+                subtitleKey={!report ? undefined : 'roomMembersPage.notAuthorized'}
```

Comment:
For objects we need to use `isEmptyObject` util instead:
```suggestion
                shouldShow={isEmptyObject(report)}
                subtitleKey={isEmptyObject(report)? undefined : 'roomMembersPage.notAuthorized'}
```
```suggestion
                shouldShow={!report}
                subtitleKey={!report ? undefined : 'roomMembersPage.notAuthorized'}
```